specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2019-05-06-preview/searchservice.json

```diff
@@ -3676,6 +3676,54 @@
         "url": "https://lucene.apache.org/core/4_10_3/analyzers-common/org/apache/lucene/analysis/pattern/PatternReplaceCharFilter.html"
       }
     },
+    "Similarity": {
+      "discriminator": "@odata.type",
+      "properties": {
+        "@odata.type": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "@odata.type"
+      ],
+      "description": "Abstract base class for similarity algorithms.",
+      "externalDocs": {
+        "url": "https://docs.microsoft.com/en-us/azure/search/index-ranking-similarity"
+      }
+    },
+    "ClassicSimilarity": {
+      "x-ms-discriminator-value": "#Microsoft.Azure.Search.ClassicSimilarity",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Similarity"
+        }
+      ],
+      "properties": {},
+      "required": [],
```

Comment:
nit: I don't think we need to provide `required` if it's empty. Possibly the same is true for `properties`.